torch/distributed/tensor/parallel/api.py

```diff
@@ -26,6 +28,9 @@
     "parallelize_module",
 ]
 
+# switch the DTensor propagator to use the caching propagator to speed up
+# the TP eager execution time.
+DTensor._propagator = _CachingPropagator(DTensor._propagator.op_to_rules)
```

Comment:
Can we have a flag to switch this on/off. Or we want to enable it by default?

Reply:
I feel we should turn this on by default, otherwise eager mode perf would be very bad, especially once we implement more complicated sharding prop. This caching logic is relatively safe as it only change the one with the exact same input specs.